android/libraries/rib-android-compose/src/main/kotlin/com/uber/rib/core/BasicComposeRouter.kt

```diff
@@ -15,7 +15,21 @@
  */
 package com.uber.rib.core
 
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+
 open class BasicComposeRouter<I : BasicInteractor<*, *>>(
   val presenter: ComposePresenter,
-  interactor: I
-) : BasicRouter<I>(interactor)
+  interactor: I,
+  val slot: MutableState<(@Composable () -> Unit)>
```

Comment:
same here, can remove mutablestate and use property delegate instead of using .value

Reply:
synced offline, motif treats type `@Composable () -> Unit` as `unitFunction0()`, as a result, we are not able to pass `@Composable () -> Unit` in parameters as a type.
`
/RIBs/android/demos/compose/build/generated/source/kaptKotlin/debug/com/uber/rib/compose/root/main/logged_in/LoggedInScopeImpl.kt: (85, 85): Type inference failed. Expected type mismatch: inferred type is () -> Unit but @Composable () -> Unit was expected
`